cmd/config/internal/commands/annotate.go

```diff
@@ -39,32 +44,68 @@ func AnnotateCommand(parent string) *cobra.Command {
 }
 
 type AnnotateRunner struct {
-	Command    *cobra.Command
-	Values     []string
-	Kind       string
-	Name       string
-	ApiVersion string
-	Namespace  string
-	Path       string
+	Command            *cobra.Command
+	Values             []string
+	Kind               string
+	Name               string
+	ApiVersion         string
+	Namespace          string
+	Path               string
+	RecurseSubPackages bool
 }
 
 func (r *AnnotateRunner) runE(c *cobra.Command, args []string) error {
 	var input []kio.Reader
 	var output []kio.Writer
+	openAPIFileName, err := ext.OpenAPIFileName()
+	if err != nil {
+		return err
+	}
 	if len(args) == 0 {
 		rw := &kio.ByteReadWriter{Reader: c.InOrStdin(), Writer: c.OutOrStdout()}
 		input = []kio.Reader{rw}
 		output = []kio.Writer{rw}
-	} else {
-		rw := &kio.LocalPackageReadWriter{PackagePath: args[0], NoDeleteFiles: true}
+
+		return handleError(c, kio.Pipeline{
+			Inputs:  input,
+			Filters: []kio.Filter{r},
+			Outputs: output,
+		}.Execute())
+	}
+
+	resourcePackagesPaths, err := pathutil.DirsWithFile(args[0], openAPIFileName, true)
```

Comment:
Is this needed? It seems like just using the existing functionality on the `LocalPackageReader` with the `InludeSubpackages` field should be sufficient?

Reply:
We will get rid of InludeSubpackages filed as a whole. So it is necessary to get all the subpackages paths.